src/states/joining_node.rs

```diff
@@ -44,19 +44,20 @@ use super::common::Testable;
 use super::Node;
 use timer::Timer;
 use types::MessageId;
-#[cfg(feature = "use-mock-crust")]
 use xor_name::XorName;
 
-/// Time (in seconds) after which a `GetNodeName` request is resent.
-const GET_NODE_NAME_TIMEOUT_SECS: u64 = 60;
+/// Time (in seconds) after which a `GetNameRange` request is resent.
+const GET_NAME_RANGE_TIMEOUT_SECS: u64 = 60;
+/// Time (in seconds) for the duration to compute a valid new id.
+const COMPUTING_IDENTITY_TIMEOUT_SECS: u64 = 30;
```

Comment:
Shorter timeout for a job requiring actual work than for a routing response? What if some network packets get dropped — it has to start again?